### PR TITLE
[Batch Infer V2] Update `model_path` type to `Union[str, mlrun.DataItem]`

### PR DESCRIPTION
**Pull request by Eyal-Danieli**
This PR includes:

- change `model_path` type from `str` to `Union[str, mlrun.DataItem]`. Then, the user can pass the model as an input. That means, he can use the UI widget for selecting available paths. 
- remove `DatasetType` (UI cannot recognize `DatasetType` object)

Related JIRAs: [ML-4640](https://jira.iguazeng.com/browse/ML-4640,) [ML-4639](https://jira.iguazeng.com/browse/ML-4639)

### CODE REVIEW

File: batch_inference_v2/batch_inference_v2.py [hunk 1]
```diff
@@ -32,9 +32,7 @@
 from mlrun.frameworks.auto_mlrun import AutoMLRun
 
 # A union of all supported dataset types:
-DatasetType = Union[
-    mlrun.DataItem, list, dict, pd.DataFrame, pd.Series, np.ndarray, Any
-]
+DatasetType = Union[list, dict, pd.DataFrame, pd.Series, np.ndarray, Any]
```
Comment (assaf758) [inline batch_inference_v2/batch_inference_v2.py:35]: I think it's better to reomve DatasetType - just use "list, dict, pd.DataFrame, pd.Series, np.ndarray" instead.

File: batch_inference_v2/batch_inference_v2.py [hunk 2]
```diff
@@ -121,8 +119,9 @@ def infer(
     :param dataset:                                 The dataset to infer through the model. Can be passed in `inputs` as either a
                                                     Dataset artifact / Feature vector URI. Or, in `parameters` as a list, dictionary or
                                                     numpy array.
-    :param model_path:                              The model Store path. If `endpoint_id` of existing model endpoint is provided,
-                                                    make sure that it has a similar model store path, otherwise the drift analysis
+    :param model_path:                              The model Store path. Can be provided as an input or as a parameter.
```
Comment (assaf758) [inline batch_inference_v2/batch_inference_v2.py:122]: Can be provided as an input (DataItem) or as a parameter (string)

File: batch_inference_v2/batch_inference_v2.py [hunk 3]
```diff
@@ -158,15 +157,19 @@ def infer(
...
+                                                    Can be provided as an input or as a parameter. The default chosen sample
+                                                    set will always be the one who is set in the model artifact itself.
 
     raises MLRunInvalidArgumentError: if both `model_path` and `endpoint_id` are not provided
     """
 
     # Loading the model:
     context.logger.info(f"Loading model...")
+    if isinstance(model_path, mlrun.DataItem):
+        model_path = model_path.artifact_url
```
Comment (assaf758) [inline batch_inference_v2/batch_inference_v2.py:170]: just making sure - record_results() knows how to handle artifact_url?

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: batch_inference_v2/batch_inference_v2.py (hunk 1 above)
Comment (gtopper) [reply batch_inference_v2/batch_inference_v2.py:35]: What's the point of a `Union` with `Any`? It's just equivalent to `Any`...

File: batch_inference_v2/batch_inference_v2.py (hunk 3 above)
Comment (Eyal-Danieli) [reply batch_inference_v2/batch_inference_v2.py:170]: yes, well this artifact url is basically the model store path (as a string).

File: batch_inference_v2/batch_inference_v2.py (hunk 1 above)
Comment (Eyal-Danieli) [reply batch_inference_v2/batch_inference_v2.py]: `DatasetType` has been removed, seems redundant

**Review by assaf758 — APPROVED** [review]

**Pull request merged**